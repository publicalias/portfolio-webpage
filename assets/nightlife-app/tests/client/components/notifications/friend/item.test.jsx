"use strict";

//local imports

const Item = require("../../../../../scripts/client/components/notifications/friend/item");

const { newFriend, newUserWithData } = require("../../../../../schemas");
const { testWrapper } = require("../../../test-helpers");

//global imports

const { initTestSnapshot, reactTests, withDataList } = require("redux/tests/react-tests");

//utilities

const { testShallow } = testWrapper(Item);

const getFriend = (confirmed = false) => newFriend({
  from: { id: "id-a" },
  to: { id: "id-b" },
  confirmed
});

//setup

beforeAll(reactTests.setup);
beforeEach(reactTests.inject(Item));

//item

describe("item (confirmed, from)", () => {

  const dataList = [{ user: newUserWithData({ id: "id-a" }) }, { friend: getFriend(true) }];

  const testSnapshot = withDataList(initTestSnapshot(testShallow), dataList);

  it("should match snapshot (default)", () => testSnapshot());

  it("should match snapshot (name)", () => testSnapshot(null, { friend: { to: { name: "User B" } } }));

});

describe("item (confirmed, to)", () => {

  const dataList = [null, { friend: getFriend(true) }];

  const testSnapshot = withDataList(initTestSnapshot(testShallow), dataList);

  it("should match snapshot (default)", () => testSnapshot());

  it("should match snapshot (name)", () => testSnapshot(null, { friend: { from: { name: "User A" } } }));

});

describe("item (unconfirmed, from)", () => {

  const dataList = [{ user: newUserWithData({ id: "id-a" }) }, { friend: getFriend() }];

  const testSnapshot = withDataList(initTestSnapshot(testShallow), dataList);

  it("should match snapshot (default)", () => testSnapshot());

  it("should match snapshot (name)", () => testSnapshot(null, { friend: { to: { name: "User B" } } }));

});

describe("item (unconfirmed, to)", () => {

  const dataList = [null, { friend: getFriend() }];

  const testSnapshot = withDataList(initTestSnapshot(testShallow), dataList);

  it("should match snapshot (default)", () => testSnapshot());

  it("should match snapshot (name)", () => testSnapshot(null, { friend: { from: { name: "User A" } } }));

});
