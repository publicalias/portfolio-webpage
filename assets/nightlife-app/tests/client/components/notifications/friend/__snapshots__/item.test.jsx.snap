// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`item (confirmed, from) should match snapshot (default) 1`] = `
<Item
  local={
    Object {
      "buttons": Array [
        Object {
          "handler": [MockFunction],
          "icon": "fas fa-times",
        },
      ],
      "item": Object {
        "confirmed": true,
        "date": 0,
        "from": Object {
          "id": "id-a",
          "name": "",
        },
        "hidden": Array [],
        "id": "",
        "to": Object {
          "id": "id-b",
          "name": "",
        },
      },
      "notification": <p>
        <Link
          to="/users/page/id-b"
        >
          <span
            className="u-underline"
          >
            Anonymous
          </span>
        </Link>
         accepted your friend request.
      </p>,
      "refresh": [MockFunction],
    }
  }
/>
`;

exports[`item (confirmed, from) should match snapshot (name) 1`] = `
<Item
  local={
    Object {
      "buttons": Array [
        Object {
          "handler": [MockFunction],
          "icon": "fas fa-times",
        },
      ],
      "item": Object {
        "confirmed": true,
        "date": 0,
        "from": Object {
          "id": "id-a",
          "name": "",
        },
        "hidden": Array [],
        "id": "",
        "to": Object {
          "id": "id-b",
          "name": "User B",
        },
      },
      "notification": <p>
        <Link
          to="/users/page/id-b"
        >
          <span
            className="u-underline"
          >
            User B
          </span>
        </Link>
         accepted your friend request.
      </p>,
      "refresh": [MockFunction],
    }
  }
/>
`;

exports[`item (confirmed, to) should match snapshot (default) 1`] = `
<Item
  local={
    Object {
      "buttons": Array [
        Object {
          "handler": [MockFunction],
          "icon": "fas fa-times",
        },
      ],
      "item": Object {
        "confirmed": true,
        "date": 0,
        "from": Object {
          "id": "id-a",
          "name": "",
        },
        "hidden": Array [],
        "id": "",
        "to": Object {
          "id": "id-b",
          "name": "",
        },
      },
      "notification": <p>
        You accepted 
        <Link
          to="/users/page/id-a"
        >
          <span
            className="u-underline"
          >
            Anonymous
          </span>
        </Link>
        's friend request.
      </p>,
      "refresh": [MockFunction],
    }
  }
/>
`;

exports[`item (confirmed, to) should match snapshot (name) 1`] = `
<Item
  local={
    Object {
      "buttons": Array [
        Object {
          "handler": [MockFunction],
          "icon": "fas fa-times",
        },
      ],
      "item": Object {
        "confirmed": true,
        "date": 0,
        "from": Object {
          "id": "id-a",
          "name": "User A",
        },
        "hidden": Array [],
        "id": "",
        "to": Object {
          "id": "id-b",
          "name": "",
        },
      },
      "notification": <p>
        You accepted 
        <Link
          to="/users/page/id-a"
        >
          <span
            className="u-underline"
          >
            User A
          </span>
        </Link>
        's friend request.
      </p>,
      "refresh": [MockFunction],
    }
  }
/>
`;

exports[`item (unconfirmed, from) should match snapshot (default) 1`] = `
<Item
  local={
    Object {
      "buttons": Array [
        Object {
          "handler": [MockFunction],
          "icon": "fas fa-ban",
        },
      ],
      "item": Object {
        "confirmed": false,
        "date": 0,
        "from": Object {
          "id": "id-a",
          "name": "",
        },
        "hidden": Array [],
        "id": "",
        "to": Object {
          "id": "id-b",
          "name": "",
        },
      },
      "notification": <p>
        You sent 
        <Link
          to="/users/page/id-b"
        >
          <span
            className="u-underline"
          >
            Anonymous
          </span>
        </Link>
         a friend request.
      </p>,
      "refresh": [MockFunction],
    }
  }
/>
`;

exports[`item (unconfirmed, from) should match snapshot (name) 1`] = `
<Item
  local={
    Object {
      "buttons": Array [
        Object {
          "handler": [MockFunction],
          "icon": "fas fa-ban",
        },
      ],
      "item": Object {
        "confirmed": false,
        "date": 0,
        "from": Object {
          "id": "id-a",
          "name": "",
        },
        "hidden": Array [],
        "id": "",
        "to": Object {
          "id": "id-b",
          "name": "User B",
        },
      },
      "notification": <p>
        You sent 
        <Link
          to="/users/page/id-b"
        >
          <span
            className="u-underline"
          >
            User B
          </span>
        </Link>
         a friend request.
      </p>,
      "refresh": [MockFunction],
    }
  }
/>
`;

exports[`item (unconfirmed, to) should match snapshot (default) 1`] = `
<Item
  local={
    Object {
      "buttons": Array [
        Object {
          "handler": [MockFunction],
          "icon": "fas fa-check",
        },
        Object {
          "handler": [MockFunction],
          "icon": "fas fa-ban",
        },
      ],
      "item": Object {
        "confirmed": false,
        "date": 0,
        "from": Object {
          "id": "id-a",
          "name": "",
        },
        "hidden": Array [],
        "id": "",
        "to": Object {
          "id": "id-b",
          "name": "",
        },
      },
      "notification": <p>
        <Link
          to="/users/page/id-a"
        >
          <span
            className="u-underline"
          >
            Anonymous
          </span>
        </Link>
         sent you a friend request.
      </p>,
      "refresh": [MockFunction],
    }
  }
/>
`;

exports[`item (unconfirmed, to) should match snapshot (name) 1`] = `
<Item
  local={
    Object {
      "buttons": Array [
        Object {
          "handler": [MockFunction],
          "icon": "fas fa-check",
        },
        Object {
          "handler": [MockFunction],
          "icon": "fas fa-ban",
        },
      ],
      "item": Object {
        "confirmed": false,
        "date": 0,
        "from": Object {
          "id": "id-a",
          "name": "User A",
        },
        "hidden": Array [],
        "id": "",
        "to": Object {
          "id": "id-b",
          "name": "",
        },
      },
      "notification": <p>
        <Link
          to="/users/page/id-a"
        >
          <span
            className="u-underline"
          >
            User A
          </span>
        </Link>
         sent you a friend request.
      </p>,
      "refresh": [MockFunction],
    }
  }
/>
`;
